src/main/java/com/rescuebites/api/client/services/implementations/ClientServiceImpl.java

```diff
@@ -37,15 +36,23 @@ public void createClient(CreateClientRequest createClientRequest, MultipartFile
         validateProfilePicture(profilePicture);
 
         User user = userService.findByIdOrThrowException(createClientRequest.userId());
-        List<Preference> preferences = loadPreferencesIfProvided(createClientRequest);
+        List<PreferenceType> preferences = loadPreferencesIfProvided(createClientRequest);
         Image image = imageService.uploadAndSaveImage(profilePicture);
 
         Client client = ClientMapper.toClient(createClientRequest, user, preferences, image);
         clientRepository.save(client);
     }
 
+    @Override
+    public ClientResponse getClientById(UUID clientId) {
+        Client client = clientRepository.findById(clientId)
+                .orElseThrow(() -> new RuntimeException("Cliente no encontrado con id: " + clientId));
+
+        return ClientMapper.toClientResponse(client);
```

Comment:
Esto podrías refactorizarlo así


```suggestion
        return clientRepository.findById(clientId)
                .map(client -> ClientMapper.toClientResponse(client))
                .orElseThrow(() -> new RuntimeException("Cliente no encontrado con id: " + clientId));
```